wait_all_test.go

```diff
@@ -0,0 +1,102 @@
+package asynctask_test
+
+import (
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/Azure/go-asynctask"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestWaitAll(t *testing.T) {
+	t.Parallel()
+	ctx, cancelFunc := newTestContextWithTimeout(t, 3*time.Second)
+	defer cancelFunc()
+
+	countingTsk1 := asynctask.Start(ctx, getCountingTask(10, 200*time.Millisecond))
+	countingTsk2 := asynctask.Start(ctx, getCountingTask(10, 20*time.Millisecond))
+	countingTsk3 := asynctask.Start(ctx, getCountingTask(10, 2*time.Millisecond))
+	completedTsk := asynctask.NewCompletedTask()
+
+	start := time.Now()
+	err := asynctask.WaitAll(ctx, &asynctask.WaitAllOptions{FailFast: true}, countingTsk1, countingTsk2, countingTsk3, completedTsk)
+	elapsed := time.Since(start)
+	assert.NoError(t, err)
+	// should only finish after longest task.
+	assert.True(t, elapsed > 10*200*time.Millisecond)
+}
+
+func TestWaitAllFailFastCase(t *testing.T) {
+	t.Parallel()
+	ctx, cancelFunc := newTestContextWithTimeout(t, 3*time.Second)
```

Comment:
ah that's neat, didn't know you could do that.
alternatively, gomega test matcher has a nice helper.
Eventually(...)